### PR TITLE
[Release-1.12] Pin contrib 1.12.4 and improve placement test reliability

### PR DESCRIPTION
**Pull request by berndverst**
# Description

Pin contrib 1.12.4 which adds more resilience to Redis Pubsub to allow recreating a consumer group for an active subscription if the group or stream are deleted for a variety of reasons.

Also cherry-picks #7167 to hopefully improve placement test reliability.

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -1172,9 +1172,9 @@ github.com/natefinch/lumberjack v2.0.0+incompatible/go.mod h1:Wi9p2TTF5DG5oU+6Yf
 github.com/nats-io/jwt v0.3.0/go.mod h1:fRYCDE99xlTsqUzISS1Bi75UBJ6ljOJQOAAu5VglpSg=
 github.com/nats-io/jwt v0.3.2 h1:+RB5hMpXUUA2dfxuhBTEkMOrYmM+gKIZYS1KjSostMI=
 github.com/nats-io/jwt v0.3.2/go.mod h1:/euKqTS1ZD+zzjYrY7pseZrTtWQSjujC7xjPc8wL6eU=
-github.com/nats-io/jwt/v2 v2.4.1 h1:Y35W1dgbbz2SQUYDPCaclXcuqleVmpbRa7646Jf2EX4=
+github.com/nats-io/jwt/v2 v2.5.0 h1:WQQ40AAlqqfx+f6ku+i0pOVm+ASirD4fUh+oQsiE9Ak=
 github.com/nats-io/nats-server/v2 v2.1.2/go.mod h1:Afk+wRZqkMQs/p45uXdrVLuab3gwv3Z8C4HTBu8GD/k=
-github.com/nats-io/nats-server/v2 v2.9.21 h1:2TBTh0UDE74eNXQmV4HofsmRSCiVN0TH2Wgrp6BD6fk=
+github.com/nats-io/nats-server/v2 v2.9.23 h1:6Wj6H6QpP9FMlpCyWUaNu2yeZ/qGj+mdRkZ1wbikExU=
```
Comment (berndverst) [inline go.sum:1177]: Fixes a security issue.

File: go.sum [hunk 2]
```diff
@@ -1183,8 +1183,8 @@ github.com/nats-io/nats.go v1.28.0/go.mod h1:XpbWUlOElGwTYbMR7imivs7jJj9GtK7ypv3
 github.com/nats-io/nkeys v0.1.0/go.mod h1:xpnFELMwJABBLVhffcfd1MZx6VsNRFpEugbxziKVo7w=
 github.com/nats-io/nkeys v0.1.3/go.mod h1:xpnFELMwJABBLVhffcfd1MZx6VsNRFpEugbxziKVo7w=
 github.com/nats-io/nkeys v0.3.0/go.mod h1:gvUNGjVcM2IPr5rCsRsC6Wb3Hr2CQAm08dsxtV6A5y4=
-github.com/nats-io/nkeys v0.4.4 h1:xvBJ8d69TznjcQl9t6//Q5xXuVhyYiSos6RPtvQNTwA=
-github.com/nats-io/nkeys v0.4.4/go.mod h1:XUkxdLPTufzlihbamfzQ7mw/VGx6ObUs+0bN5sNvt64=
+github.com/nats-io/nkeys v0.4.6 h1:IzVe95ru2CT6ta874rt9saQRkWfe2nFj1NtvYSLqMzY=
+github.com/nats-io/nkeys v0.4.6/go.mod h1:4DxZNzenSVd1cYQoAa8948QY3QDjrHfcfVADymtkpts=
```
Comment (berndverst) [inline go.sum:1187]: Fixes a security issue.

File: go.mod [hunk 3]
```diff
@@ -60,16 +60,16 @@ require (
 	go.uber.org/ratelimit v0.3.0
 	golang.org/x/crypto v0.14.0
 	golang.org/x/exp v0.0.0-20230905200255-921286631fa9
-	golang.org/x/net v0.15.0
+	golang.org/x/net v0.17.0
 	golang.org/x/sync v0.3.0
 	google.golang.org/genproto/googleapis/api v0.0.0-20230803162519-f966b187b2e5
 	google.golang.org/genproto/googleapis/rpc v0.0.0-20230807174057-1744710a1577
-	google.golang.org/grpc v1.57.0
+	google.golang.org/grpc v1.58.3
```
Comment (berndverst) [inline go.mod:67]: Fixes a security issue.

File: go.mod [hunk 4]
```diff
@@ -60,16 +60,16 @@ require (
...
 	google.golang.org/protobuf v1.31.0
 	gopkg.in/yaml.v3 v3.0.1
 	k8s.io/api v0.26.9
 	k8s.io/apiextensions-apiserver v0.26.3
-	k8s.io/apimachinery v0.26.9
+	k8s.io/apimachinery v0.26.10
```
Comment (berndverst) [inline go.mod:72]: Fixes a security issue in goproxy

File: go.mod [hunk 5]
```diff
@@ -311,7 +311,7 @@ require (
 	github.com/munnerz/goautoneg v0.0.0-20191010083416-a7dc8b61c822 // indirect
 	github.com/natefinch/lumberjack v2.0.0+incompatible // indirect
 	github.com/nats-io/nats.go v1.28.0 // indirect
-	github.com/nats-io/nkeys v0.4.4 // indirect
+	github.com/nats-io/nkeys v0.4.6 // indirect
```
Comment (berndverst) [inline go.mod:314]: Fixes a security issue.

**Review by artursouza — APPROVED** [review]

**Comment by berndverst** [comment]
Interestingly the unit tests pass locally on macOS but fail in CI (on placement)

**Comment by ItalyPaleAle** [comment]
> Interestingly the unit tests pass locally on macOS but fail in CI (on placement)

Probably same reason as #7167

Seems that upgrading some deps causes some timing issues in the placement test

File: go.mod [hunk 6]
```diff
@@ -60,16 +60,16 @@ require (
 	go.uber.org/ratelimit v0.3.0
 	golang.org/x/crypto v0.14.0
 	golang.org/x/exp v0.0.0-20230905200255-921286631fa9
-	golang.org/x/net v0.15.0
+	golang.org/x/net v0.17.0
 	golang.org/x/sync v0.3.0
 	google.golang.org/genproto/googleapis/api v0.0.0-20230803162519-f966b187b2e5
 	google.golang.org/genproto/googleapis/rpc v0.0.0-20230807174057-1744710a1577
-	google.golang.org/grpc v1.57.0
+	google.golang.org/grpc v1.57.1
```
Comment (berndverst) [inline go.mod:67]: Fixes a security issue in grpc

**Comment by berndverst** [comment]
I made a new contrib version that uses a 1.57.X patch version (some minor version has what we had) to address the GRPC vulnerability. Hopefully this will not result in build failures.

**Comment by berndverst** [comment]
Security issues were now fixed in a previous PR - #7178

**Comment by berndverst** [comment]
This PR only fixes the REDIS try behavior

**Review by artursouza — APPROVED** [review]

**Pull request merged**